tests/h/api/models/annotation_test.py

```diff
@@ -83,6 +83,94 @@ def test_acl_group_shared():
     assert actual == expect
 
 
+def test_setting_extras_inline_is_persisted(db_session):
+    """
+    In-place changes to Annotation.extra should be persisted.
+
+    Setting an Annotation.extra value in-place:
+
+        my_annotation.extra['foo'] = 'bar'
+
+    should be persisted to the database.
+
+    """
+    annotation = Annotation(userid='fred')
+    annotation.extra = {}  # FIXME: There should be a default value for this.
+
+    # Add the annotation to the db so that annotation.id gets generated.
```

Comment:
Possibly leaning a little towards the `# Add 1 to i` style of commenting here ;)